Aztec/Classes/TextKit/TextStorage.swift

```diff
@@ -363,6 +366,48 @@ open class TextStorage: NSTextStorage {
 }
 
 
+// MARK: - Header Font Attribute Fixes
+//
+private extension TextStorage {
+
+    /// Ensures that the TextFont matches the new HeaderLevel that's about to be applied.
+    ///
+    /// - Parameters:
+    ///   - attrs: NSAttributedString attributes that are about to be applied.
+    ///   - range: Range that's about to be affected by the new Attributes collection.
+    ///
+    /// - Returns: Collection of attributes with the Font Attribute corrected, if needed.
+    ///
+    func ensureFontAttributeIsValid(beforeApplying attrs: [String: Any], at range: NSRange) -> [String: Any] {
```

Comment:
In line with my previous comment, maybe rename this to something `ensureMatchingFontAndParagraphHeaderStyles()`?

The problem with `ensureFontAttributeIsValid` is that it doesn't really describe what's happening.